File: src/inspect_ai/_view/www/src/samples/SampleLimit.mjs
//

/**
 * Formats a limit message
 *
 * @param {import("../types/log").Type11} [type] - The limit type
 * @returns {string} The limit message
 */
export const sampleLimitMessage = (type) => {
  switch (type) {
    case "operator":
      return "Sample terminated due to operator limit.";
    case "message":
      return "Sample terminated due to message limit.";
    case "time":
      return "Sample terminated due to time limit.";
    case "token":
      return "Sample terminated due to token limit.";
    case "context":
      return "Sample terminated due to context limit.";
    default:
      return undefined;
  }
};

File: src/inspect_ai/event/_sample_limit.py
from typing import Literal

from pydantic import Field

from inspect_ai.event._base import BaseEvent


class SampleLimitEvent(BaseEvent):
    """The sample was unable to finish processing due to a limit"""

    event: Literal["sample_limit"] = Field(default="sample_limit")
    """Event type."""

    type: Literal["message", "time", "working", "token", "cost", "operator", "custom"]
    """Type of limit that halted processing"""

    message: str
    """A message associated with this limit"""

    limit: float | None = Field(default=None)
    """The limit value (if any)"""
